zebra-state/src/lib.rs

```diff
@@ -31,6 +31,19 @@ use zebra_chain::{
 pub mod in_memory;
 pub mod on_disk;
 
+/// The maturity threshold for transparent coinbase outputs.
+///
+/// A transaction MUST NOT spend a transparent output of a coinbase transaction
+/// from a block less than 100 blocks prior to the spend. Note that transparent
+/// outputs of coinbase transactions include Founders' Reward outputs.
+const MIN_TRASPARENT_COINBASE_MATURITY: BlockHeight = BlockHeight(100);
+
+/// The maximum chain reorganisation height.
+///
+/// Allowing reorganisations past this height could allow double-spends of
+/// coinbase transactions.
```

Comment:
I don't think this is accurate, we could for example recreate and invert the changes to the UTXO set from the blocks that are being removed from the disk and revert those in the UTXO set before appyling the new set of UTXO deltas.

Reply:
It's true that Zebra could implement arbitrary rollbacks, but that would cause security issues, and make the code more complex.

Here are some examples of rollback attacks:
1. Spend the coinbase transaction on Bitcoin, rollback, spend it again
2. Create an alternate chain where you mine all the blocks, and own the coinbase outputs
3. Rollback 100 blocks, rollback another 100 blocks, repeat and destroy the chain

Since arbitrary rollbacks allow these attacks, we don't need to allow for them in our design. Instead, we can match the zcashd rollback limit of 99 blocks.

Historically, Zcash has a few 1-block side-chains, and has historically had splits up to 7 blocks on isolated parts of the network. So there's no practical reason to allow for anything more than 99 blocks.